apps/dashboard/src/components/create-workflow-button.tsx

```diff
@@ -36,8 +36,9 @@ export const CreateWorkflowButton = (props: CreateWorkflowButtonProps) => {
   const navigate = useNavigate();
   const { currentEnvironment } = useEnvironment();
   const [isOpen, setIsOpen] = useState(false);
+  // TODO: Move to a use-create-workflow.ts hook
   const { mutateAsync, isPending } = useMutation({
-    mutationFn: async (data: CreateWorkflowDto) => createWorkflow(data),
+    mutationFn: async (workflow: CreateWorkflowDto) => createWorkflow({ environment: currentEnvironment!, workflow }),
```

Comment:
In a follow-up PR, `currentEnvironment` will never be undefined or null to avoid the `!` or `?._id` discomfort.